modules/unsupported/mongodb/src/main/java/org/geotools/data/mongodb/FilterToMongo.java

```diff
@@ -403,9 +403,14 @@ public Object visit(BBOX filter, Object extraData) {
         Object e1 = filter.getExpression1().accept(this, Geometry.class);
 
         Envelope envelope = filter.getExpression2().evaluate(null, Envelope.class);
-
-        DBObject dbo = BasicDBObjectBuilder.start().push("$geoIntersects")
-                .add("$geometry", geometryBuilder.toObject(envelope)).get();
+                
+        DBObject geometryDBObject = geometryBuilder.toObject(envelope);
+        addCrsToGeometryDBObject(geometryDBObject);
+        DBObject dbo = BasicDBObjectBuilder
+        		.start()
+        		.push("$geoIntersects")
+                .add("$geometry", geometryDBObject)
+                .get();
```

Comment:
Sorry, that was all me :).   I made sure I had the templates/formatter imported and did a re-format.  When you get a chance take a look and let me know if it looks better.  

Thanks!